Cython/Compiler/ExprNodes.py

```diff
@@ -263,6 +263,20 @@ def translate_cpp_exception(code, pos, inside, py_result, exception_value, nogil
     code.putln(code.error_goto(pos))
     code.putln("}")
 
+def needs_cpp_exception_conversion(node):
+    # This is only called after confirming that node.exception_check == "+"
+    if node.exception_value is None:
+        return True
+    # exception_value can be a NameNode
+    # (in which case it's used as a handler function and no conversion is needed)
+    if node.exception_value.is_name:
+        return False
+    # or a CharNode with a value of "*"
+    if isinstance(node.exception_value, CharNode) and node.exception_value.value == "*":
```

Comment:
This might be enough – assuming we don't get exception value nodes with uninitialised `constant_result` values from … somewhere. But that would be a bug, I guess. :)
```suggestion
    if node.exception_value.constant_result == "*":
```

Reply:
`constant_result` isn't calculated because of `SkipDeclarations` in `ConstantFolding`. It ends up being under several layers of declarations, so I'll just revert this change rather than try to get `constant_result` calculated.